cv_bridge/src/module.hpp

```diff
@@ -33,16 +33,9 @@ int convert_to_CvMat2(const PyObject* o, cv::Mat& m);
 
 PyObject* pyopencv_from(const cv::Mat& m);
 
-#if PYTHON3
 static int do_numpy_import( )
 {
     import_array( );
```

Comment:
Confusingly enough, `import_array` is implemented as a macro - and with newer Python/numpy it actually has a `return NULL` statement inside (I believe [`numpy`'s API generator](https://github.com/numpy/numpy/blob/c1f1bc9ce8e4e2936e80d9bfafc3d8e03237a84b/numpy/core/code_generators/generate_numpy_api.py#L112) may give some insight). What's worse is that the statement is not always called, which leads to some weird side effects - I've had a reasonably recent compiler optimize the whole function call out since not all paths returned a value.

All this is to say, please change the function signature to return `void *` and have a `return nullptr` or something after the `import_array()` :)

Reply:
https://github.com/ros-perception/vision_opencv/pull/323/commits/060a591116a4d3709031f54f2303e8704e8e4dbb